aws/resource_aws_cloudformation_stack_set.go

```diff
@@ -250,45 +250,43 @@ func resourceAwsCloudFormationStackSetUpdate(d *schema.ResourceData, meta interf
 func resourceAwsCloudFormationStackSetDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).cfconn
 
-	input := &cloudformation.DeleteStackSetInput{
-		StackSetName: aws.String(d.Id()),
-	}
-
 	log.Printf("[DEBUG] Deleting CloudFormation StackSet: %s", d.Id())
-	_, err := conn.DeleteStackSet(input)
-
-	if isAWSErr(err, cloudformation.ErrCodeStackSetNotFoundException, "") {
-		return nil
-	}
-
+	err := deleteCloudFormationStackSet(conn, d.Id())
 	if err != nil {
 		return fmt.Errorf("error deleting CloudFormation StackSet (%s): %s", d.Id(), err)
 	}
 
 	return nil
 }
 
-func listCloudFormationStackSets(conn *cloudformation.CloudFormation) ([]*cloudformation.StackSetSummary, error) {
+func deleteCloudFormationStackSet(conn *cloudformation.CloudFormation, name string) error {
+	_, err := conn.DeleteStackSet(&cloudformation.DeleteStackSetInput{
+		StackSetName: aws.String(name),
+	})
+	if isAWSErr(err, cloudformation.ErrCodeStackSetNotFoundException, "") {
+		return nil
+	}
+	return err
+}
+
+// generate
+func listAllCloudFormationStackSetsPages(conn *cloudformation.CloudFormation, fn func(*cloudformation.ListStackSetsOutput, bool) bool) error {
```

Comment:
Here we're creating our own `List*Pages()` from the `List*()` function the AWS API provides. This can be easily generated, since it follows a standard pattern, and could be swapped out if/when the API adds a paged function.